dynamiqs/solvers/ode/adaptive_integrator.py

```diff
@@ -154,7 +155,7 @@ def build_tableau(self, target):
         return alpha, beta, csol5, csol5 - csol4
 
     def step(
-        self, f0: Tensor, y0: Tensor, t0: float, dt: float
+        self, f0: Tensor, y0: Tensor, t0: Tensor, dt: float
```

Comment:
We should maybe add a comment saying that `t0` is a one-element tensor. Another option is to convert `t_save` to a numpy array to have proper floats. I quite like this option, because it makes clear that no gradient is computed w.r.t. time variable. It would also avoid the `dt = ts.item() - t` in `adaptive_solver.py`. And finally going through the codebase this would fix the last typing mistake we need, in `.solver.py` for the `def next_tsave(self) -> float:` function that currently returns a tensor.